app/views/shared/_address.html.erb

```diff
@@ -0,0 +1,28 @@
+<%= form.fields_for :address, address do |address_form| %>
+  <div class="mb-3">
+    <%= address_form.label :address_line_1, class:"form-label"%>
+    <%= address_form.text_field :address_line_1, class:"form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= address_form.label :address_line_2, class:"form-label"%>
+    <%= address_form.text_field :address_line_2, class:"form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= address_form.label :city, class:"form-label"%>
+    <%= address_form.text_field :city, class:"form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= address_form.label :state, class:"form-label"%>
+    <%= address_form.text_field :state, class:"form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= address_form.label :pincode, class:"form-label"%>
+    <%= address_form.number_field :pincode, class:"form-control" %>
+  </div>
+  <% if address.id && address.addressable_type == 'User' %>
```

Comment:
Create a method on user and address or wherever you are going to use address.
Check there if the address is destroyable or not.
Then call that method on `addressable`

Reply:
Done